### PR TITLE
Stash issue severity threshold settings

### PR DESCRIPTION
**Pull request by Zhabha**
Prevents comments for low severity issues to be added

### CODE REVIEW

File: src/test/java/org/sonar/plugins/stash/StashRequestFacadeTest.java [hunk 1]
```diff
@@ -368,6 +368,7 @@ public void testGetStashPullRequestIdThrowsException() throws StashConfiguration
 
   @Test
   public void testPostCommentPerIssue() throws Exception {
+    when(config.getIssueSeverityThreshold()).thenReturn(Severity.INFO);
```
Comment (t-8ch) [inline src/test/java/org/sonar/plugins/stash/StashRequestFacadeTest.java]: Can you move all these calls to the setUp() method?

Comment (Zhabha) [reply src/test/java/org/sonar/plugins/stash/StashRequestFacadeTest.java]: moved calls to setUp()

File: README.md [hunk 2]
```diff
@@ -54,6 +54,8 @@ Go to Stash general settings screen on SonarQube server to fill:
 
 **Stash issue threshold** (sonar.stash.issue.threshold): To limit the number of issue pushed to Stash.
 
+**Stash issue severity threshold** (sonar.stash.issue.severity.threshold): To push issues of specified or higher severity to Stash. Affects only diff comments, overview will still contain all severities. By default, all diff issues are pushed (threshold: INFO).
```
Comment (Zhabha) [inline README.md]: readme updated to indicate that this settings will affect only the diff comments, not the summary one.

**Comment by t-8ch** [comment]
Thanks! (I will take another look tomorrow, and then most probably merge it)

File: README.md (hunk 2 above)
Comment (t-8ch) [reply README.md]: I think we can remove the `(threshold: INFO)`, as it is implicit.

File: src/main/java/org/sonar/plugins/stash/StashPlugin.java [hunk 3]
```diff
@@ -124,6 +125,14 @@ public void define(Context context) {
                           .subCategory(CONFIG_PAGE_SUB_CATEGORY_STASH)
                           .onQualifiers(Qualifiers.PROJECT)
                           .defaultValue(DEFAULT_STASH_THRESHOLD_VALUE).build(),
+        PropertyDefinition.builder(STASH_ISSUE_SEVERITY_THRESHOLD)
+                          .name("Stash issue severity Threshold")
+                          .description("Only push issues with the same or higher severity to Stash server")
+                          .type(PropertyType.SINGLE_SELECT_LIST)
+                          .subCategory(CONFIG_PAGE_SUB_CATEGORY_STASH)
+                          .onQualifiers(Qualifiers.PROJECT)
+                          .defaultValue(Severity.INFO)
```
Comment (t-8ch) [inline src/main/java/org/sonar/plugins/stash/StashPlugin.java]: Personally I would prefer to use `SEVERITY_LIST.get(0)` as the default value, what is your opiniion?

File: src/main/java/org/sonar/plugins/stash/StashPlugin.java [hunk 4]
```diff
@@ -124,6 +125,14 @@ public void define(Context context) {
                           .subCategory(CONFIG_PAGE_SUB_CATEGORY_STASH)
                           .onQualifiers(Qualifiers.PROJECT)
                           .defaultValue(DEFAULT_STASH_THRESHOLD_VALUE).build(),
+        PropertyDefinition.builder(STASH_ISSUE_SEVERITY_THRESHOLD)
+                          .name("Stash issue severity Threshold")
+                          .description("Only push issues with the same or higher severity to Stash server")
```
Comment (t-8ch) [inline src/main/java/org/sonar/plugins/stash/StashPlugin.java]: This should match the description in the readme

File: README.md (hunk 2 above)
Comment (t-8ch) [reply README.md]: I would also prefer the wording:
`Minimum issue severity to create diff-view comments for.` (or something like this)

File: src/main/java/org/sonar/plugins/stash/StashPlugin.java (hunk 3 above)
Comment (Zhabha) [reply src/main/java/org/sonar/plugins/stash/StashPlugin.java]: Good idea, should there be a lower severity added in the future, it'll still pick the lowest one properly.

**Comment by Zhabha** [comment]
Documentation and default severity updated according proposal.

**Comment by t-8ch** [comment]
Thanks!

**Pull request merged**